tmk_core/protocol/arm_atsam/main_arm_atsam.c

```diff
@@ -160,6 +155,75 @@ void send_consumer(uint16_t data)
 #endif //EXTRAKEY_ENABLE
 }
 
+uint8_t g_drvid;
+
+void main_subtask_usb_state(void)
+{
+    if (usb_state == USB_STATE_POWERDOWN)
+    {
+        uint32_t timer_led = timer_read32();
+
+        led_on;
+        if (led_enabled)
+        {
+            for (g_drvid=0;g_drvid<ISSI3733_DRIVER_COUNT;g_drvid++)
+            {
+                I2C3733_Control_Set(0);
+            }
+        }
+        while (usb_state == USB_STATE_POWERDOWN)
+        {
+            if (timer_read32() - timer_led > 1000) led_off; //Good to indicate went to sleep, but only for a second
+        }
+        if (led_enabled)
+        {
+            for (g_drvid=0;g_drvid<ISSI3733_DRIVER_COUNT;g_drvid++)
+            {
+                I2C3733_Control_Set(1);
+            }
+        }
+        led_off;
+    }
+}
+
+void main_subtask_led(void)
+{
+    led_matrix_task();
+}
+
+uint64_t g_next_5v_checkup = 0;
+void main_subtask_power_check(void)
+{
+    if (CLK_get_ms() > g_next_5v_checkup)
+    {
+        g_next_5v_checkup = CLK_get_ms() + 5;
+
+        v_5v = adc_get(ADC_5V);
+        v_5v_avg = 0.9 * v_5v_avg + 0.1 * v_5v;
+
+        gcr_compute();
+    }
+}
+
+uint64_t g_next_usb_checkup = 0;
+void main_subtask_usb_extra_device(void)
+{
+    if (CLK_get_ms() > g_next_usb_checkup)
```

Comment:
Make `g_next_usb_checkup` a static local variable: put `static uint64_t next_usb_checkup;` inside the function body above this line (presumably `g_` stands for _global_, so it's omitted).